helm/aws-operator/values.yaml

```diff
@@ -1,20 +1,24 @@
 image:
   registry: "quay.io"
-  repository: "giantswarm/aws-operator"
-  tag: "[[ .Version ]]"
+  repository: "giantswarm/{{ .Chart.Name }}"
+  tag: "{{ .Chart.Version }}"
 pod:
   user:
     id: 1000
   group:
     id: 1000
 project:
-  name: "aws-operator"
+  name: "{{ .Chart.Name }}"
+# Resource names are trimmed to 47 characters. Kubernetes allows 63 characters
+# limit for resource names. When pods for deployments are created they have
+# additional 16 characters suffix, e.g. "-957c9d6ff-pkzgw" and we want to have
+# room for those suffixes.
 resource:
   default:
-    name:  aws-operator-v[[ .Version | replace "." "-" ]]
+    name:  "{{ .Release.Name | trim 47 }}"
     namespace: "giantswarm"
   psp:
-    name: aws-operator-psp-v[[ .Version | replace "." "-" ]]
+    name: "{{ .Release.Name | trim 47 }}-psp"
   pullSecret:
-    name: aws-operator-pull-secret-v[[ .Version | replace "." "-" ]]
+    name: "{{ .Release.Name | trim 47 }}-pull-secret"
     namespace: "giantswarm"
```

Comment:
What I like is that we now use `[[` templating only in `Chart.yaml`. In a single place.

Reply:
There is also no `aws-operator` in values.yaml making it even more generic.

Reply:
So what is the result of the trimmed name? Can we have examples for a couple of operators like `aws-operator` and `prometheus-config-controller`?

Reply:
Helm release name becomes the name of the App CR. So it will be the name of the App CR trimmed to 47 characters. E.g. `aws-operator-v2-0-0` or `prometheus-config-controller-v1-1-2-abc12de`. Depends how release-operatror will name managed App CRs. So the control over that is level above. We had problems with versions with full git SHA. E.g. `prometheus-config-controller-v1-0-0-28a55852c391fce54ca13629aa99caca1fe12824` would be too long and that would be trimmed to `prometheus-config-controller-v1-0-0-28a55852c39`. But this is only for testing. In release-operator we most likely will use first 7 characters of the SHA for testing components. For released components whole name should always fit. Even that `prometheus-config-controller-v1000-1000-1000` is still 44 characters.